extensions/amp-fit-text/0.2/fit-text.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import * as styles from './fit-text.css';
+import {omit} from '../../../src/utils/object';
+import {px, setStyle} from '../../../src/style';
+import {useCallback, useEffect, useRef, useState} from '../../../src/preact';
+
+const {LINE_HEIGHT_EM_} = styles;
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function FitText(props) {
+  const {
+    'children': children,
+    'minFontSize': minFontSize = 6,
+    'maxFontSize': maxFontSize = 72,
+    'width': width = props['style']['width'] || '100%',
+    'height': height = props['style']['height'] || '100%',
+  } = props;
+  const rest = omit(props, [
+    'children',
+    'minFontSize',
+    'maxFontSize',
+    'width',
+    'height',
+  ]);
+  const contentRef = useRef(null);
+  const measurerRef = useRef(null);
+  const [wrapperStyle, setWrapperStyle] = useState(/** @type {Object} */ ({}));
+
+  const resizer = useCallback(() => {
+    return (maxHeight, maxWidth) => {
```

Comment:
Can you explain why we return a closure here? Couldn't we use `useCallback((maxHeight, maxWidth) => { … })`, and change all the `resizer()(height, width)` to `resizer()(height, width)`?

Reply:
I've changed it to the way you described but it's still not clear to me that the callback can take parameters 😅  If we are working from the [same signature](https://github.com/ampproject/amphtml/pull/28169#discussion_r421560625) that says it only returns `void` then I don't think it can (?) but then it also can't return what it's returning here.

Reply:
Yah, I think I was wrong there. Let's use this single callback, without an intermediate closure.